crates/holochain/tests/tests/websocket/mod.rs

```diff
@@ -152,14 +155,63 @@ async fn call_zome() {
     let response = check_timeout(response, 3000).await.unwrap();
     assert_matches!(response, AdminResponse::AppEnabled { .. });
 
+    // Attach App Interface
+    let app_port = attach_app_interface(&admin_tx, None).await;
+
+    let (app_tx, app_rx) = websocket_client_by_port(app_port).await.unwrap();
+    let _app_rx = WsPollRecv::new::<AppResponse>(app_rx);
+    authenticate_app_ws_client(app_tx.clone(), admin_port, "test".to_string()).await;
+
+    // Authentication of zome call should fail with invalid signature.
+    let zome_name = TestWasm::Foo.coordinator_zome_name();
+    let fn_name = FunctionName("foo".into());
+    let (nonce, expires_at) = holochain_nonce::fresh_nonce(Timestamp::now()).unwrap();
+    let mut zome_call_params = ZomeCallParams {
+        provenance: cell_id.agent_pubkey().clone(),
+        cell_id: cell_id.clone(),
+        zome_name: zome_name.clone(),
+        fn_name: fn_name.clone(),
+        cap_secret: None,
+        payload: ExternIO::encode(()).unwrap(),
+        nonce,
+        expires_at,
+    };
+    let bytes = encode(&zome_call_params).unwrap();
+    let signature = fixt!(Signature);
+    let request = AppRequest::CallZome(Box::new(ZomeCallParamsSigned::new(
+        bytes,
+        signature.clone(),
+    )));
+    let response = app_tx.request(request);
+    let response = check_timeout(response, 6000).await.unwrap();
+    assert_matches!(
```

Comment:
Should this be separate tests? It's doing a couple of different things and asserting.

I'd prefer a common setup function and shorter tests

Reply:
Cool, I'll do that.